### PR TITLE
detect correct main vertical and horizontal axis positions

### PR DESCRIPTION
**Pull request by theiliad**
Just had to optimize the logic there a bit

### CODE REVIEW

File: packages/core/src/services/scales-cartesian.ts [hunk 1]
```diff
@@ -214,10 +214,14 @@ export class CartesianScales extends Service {
 
 	protected findMainVerticalAxisPosition() {
 		const options = this.model.getOptions();
-		const axisOptions = Tools.getProperty(options, "axes");
+		const axesOptions = Tools.getProperty(options, "axes");
 
 		// If right axis has been specified as `main`
-		if (Tools.getProperty(axisOptions, AxisPositions.RIGHT, "main") === true) {
+		if (
+			(Tools.getProperty(axesOptions, AxisPositions.LEFT) === null &&
```
Comment (zvonimirfras) [inline packages/core/src/services/scales-cartesian.ts:221]: Indentation makes it looks like the grouping is opposite from what it is.
Also braces around the && are not actually needed.

Comment (theiliad) [reply packages/core/src/services/scales-cartesian.ts:221]: could you write the if-statement as you're imagining it please? I didn't really understand your comment there. Also we've got #617 which adds prettier

Comment (zvonimirfras) [reply packages/core/src/services/scales-cartesian.ts:221]: I really dislike prettier and don't think we should add it.

Comment (zvonimirfras) [reply packages/core/src/services/scales-cartesian.ts:221]: There are so many reasonable ways to do it, feel free to pick one that suggests the right operator priority.

Comment (theiliad) [reply packages/core/src/services/scales-cartesian.ts:221]: the point of prettier is to get rid of our different opinions

File: packages/core/src/services/scales-cartesian.ts [hunk 2]
```diff
@@ -226,10 +230,14 @@ export class CartesianScales extends Service {
...
-		const axisOptions = Tools.getProperty(options, "axes");
+		const axesOptions = Tools.getProperty(options, "axes");
 
 		// If top axis has been specified as `main`
-		if (Tools.getProperty(axisOptions, AxisPositions.TOP, "main") === true) {
+		if (
+			(Tools.getProperty(axesOptions, AxisPositions.BOTTOM) === null &&
+				Tools.getProperty(axesOptions, AxisPositions.TOP) !== null) ||
+				Tools.getProperty(axesOptions, AxisPositions.TOP, "main") === true
+		) {
```
Comment (zvonimirfras) [inline packages/core/src/services/scales-cartesian.ts:240]: For maximum verbosity, you could go with
```suggestion
		if ((
				Tools.getProperty(axesOptions, AxisPositions.BOTTOM) === null
				&& Tools.getProperty(axesOptions, AxisPositions.TOP) !== null
			)
			|| Tools.getProperty(axesOptions, AxisPositions.TOP, "main") === true
		) {
```
or something along those lines

**Pull request merged**